From df0eb7a3d01772a92993e136f187c6cbeac229c5 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@example.com>
Date: Wed, 9 Apr 2014 21:37:16 +0200
Subject: [PATCH 1/3] Move print functionality of is_prime

Moves print functionality from is_prime to its own function
---
 math.c | 25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/math.c b/math.c
index d393b41..590d89a 100644
--- a/math.c
+++ b/math.c
@@ -4,6 +4,7 @@
  */
 #include <stdio.h>
 #include <math.h>
+#include <stdbool.h>
 
 /* Short pi calculation */
 void pi(void){
@@ -13,7 +14,9 @@ void pi(void){
 }
 
 /* More funny math
-   Check if number is a prime no
+   Returns:
+   0 if not prime
+   1 if prime
 */
 int is_prime(int n){
     if (n<2){
@@ -26,20 +29,30 @@ int is_prime(int n){
     int i = 2;
     while (i <= max) {
         if (n % i == 0){
-            printf("%d is not a prime number\n", n);
-            return 0;
+            return false;
         }
         i++;
     }
-    printf("%d is a prime number!\n", n);
+    return true;
+}
 
-    return 0;
+/* Test if prime and write result */
+void test_no(int number){
+    if(is_prime(number)) {
+        printf("%d is a prime number!\n", number);
+    }
+    else {
+        printf("%d is not a prime number!\n", number);
+
+    }
 }
 
 int main (void){
     pi();
 
-    is_prime(223);
+    test_no(223);
+    test_no(3863);
+    test_no(583);
 
     return 0;
 }
-- 
2.37.2.windows.2

